sci-mathematics/giac/files/giac-1.9.0.995-glibcxx-assertions.patch

```diff
@@ -0,0 +1,21 @@
+From Michael Orlitzky <michael@orlitzky.com> in the original version
+of the patch:
+
+This was added upstream when it was pointed out that the codebase
+contains undefined behavior that triggers glibcxx's assertions.
+Disabling the assertions without fixing the undefined behavior is not
+a satisfactory solution, so we drop the override.
+
+diff --git a/configure.ac b/configure.ac
+index 96b0223..cdc5618 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -48,7 +48,7 @@ AC_CHECK_FUNCS(tgammaf)
+ dnl avoid "security" checks for vectors is also included in first.h
```

Comment:
Can you please also comment out the `_GLIBCXX_ASSERTIONS` definition in `first.h`? I guess it was added in this release.

Reply:
I guess I should explain: the build should pass with `_GLIBCXX_ASSERTIONS` enabled (defined to 1). See e.g. https://bugs.gentoo.org/show_bug.cgi?id=876895

Giac has problems with it because Parisse insists that he should be able to ask for the (n+1)st element of a container that has n elements. It works in C because arrays are the only container type and arrays are contiguous, but the C++ standard does not allow it for the containers in the standard library (there's no promise that the elements are contiguous in memory).

Disabling `_GLIBCXX_ASSERTIONS` is not a good solution to the problem because the code it fails on is actually undefined behavior, glibcxx is just letting you know.

Reply:
I said "the build should pass," but the tests should pass. The assertions crash it at runtime, not build time.

Reply:
> I said "the build should pass," but the tests should pass. The assertions crash it at runtime, not build time.

Understood. I will not comment further on the quality of giac's code...